tasks/src/main/scala/dagr/tasks/gatk/GatkTask.scala

```diff
@@ -49,18 +74,35 @@ abstract class GatkTask(val walker: String,
   requires(Cores(1), Memory("4g"))
 
   override def args: Seq[Any] = {
-    val buffer = ListBuffer[Any]()
-    buffer.appendAll(jarArgs(this.gatkJar, jvmMemory=this.resources.memory))
-    buffer.append("-T", this.walker)
+    val buffer  = ListBuffer[Any]()
+    val jvmArgs = if (gatkMajorVersion >= 4) bamCompression.map(c => s"-Dsamjdk.compression_level=$c") else None
+    buffer.appendAll(jarArgs(this.gatkJar, jvmMemory=this.resources.memory, jvmArgs=jvmArgs))
+
+    if (gatkMajorVersion < 4) buffer += "-T"
+    buffer += this.walker
+
+    if (gatkMajorVersion < 4) bamCompression.foreach(c => buffer.append("--bam_compression", c))
+
     buffer.append("-R", this.ref.toAbsolutePath.toString)
     intervals.foreach(il => buffer.append("-L", il.toAbsolutePath.toString))
-    bamCompression.foreach(c => buffer.append("--bam_compression", c))
+
     addWalkerArgs(buffer)
     buffer.toSeq
   }
 
   /** Can be overridden to use a specific GATK jar. */
   protected def gatkJar: Path = configure[Path](GatkTask.GatkJarPathConfigKey)
 
+  /** The version of GATK being used by this task. */
+  protected lazy val gatkMajorVersion: Int = {
+    val jar = gatkJar
+    if (jar == GatkTask.GatkJarPath) GatkTask.GatkMajorVersion
+    else GatkTask.majorVersion(jar)
+  }
```

Comment:
Ditto about this being something to bake into `JarTask`.  We could also have a trait too, since not everything is a JAR.